app/controllers/pages_controller.rb

```diff
@@ -0,0 +1,15 @@
+class PagesController < ApplicationController
+  def show
+    if valid_page?
+      render template: "pages/#{params[:page]}"
+    else
+      render file: "#{Rails.root}/public/404", layout: false, status: :not_found
+    end
+  end
+
+  private
+
+  def valid_page?
+    File.exist?(Pathname.new(Rails.root + "app/views/pages/#{params[:page]}.html.slim"))
```

Comment:
@marderer what the purpose of this validation? are you checking whether view file exists?

Reply:
Yes

Reply:
@marderer let's use one-route-one-action way for information pages. Such dynamic way isn't easy to support, especially when you have different preloaded data on different pages.

For me the easiest way here will be:
```ruby
def show
end
```

instead of
```ruby
def show
  if valid_page?
    render template: "pages/#{params[:page]}"
  else
    render file: "#{Rails.root}/public/404", layout: false, status: :not_found
  end
end

private

def valid_page?
  File.exist?(Pathname.new(Rails.root + "app/views/pages/#{params[:page]}.html.slim"))
end
```

See: in suggested solution, we have only one page in `show` action, for others, we'll create separate actions